File: core/src/main/java/com/jinloes/claudereviews/services/stream/ContentBlock.java
package com.jinloes.claudereviews.services.stream;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.Map;
import java.util.Optional;
import lombok.AccessLevel;
import lombok.Data;
import lombok.Getter;

@Data
@Getter(AccessLevel.NONE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class ContentBlock {

    private String type;
    private String name;
    private Map<String, Object> input;
    private String text;
    private String thinking;

    public String type() {
        return type;
    }

    public Optional<String> name() {
        return Optional.ofNullable(name);
    }

    public Optional<Map<String, Object>> input() {
        return Optional.ofNullable(input);
    }

    public Optional<String> text() {
        return Optional.ofNullable(text);
    }

    public Optional<String> thinking() {
        return Optional.ofNullable(thinking);
    }
}

File: core/src/jvmMain/kotlin/com/jinloes/prpilot/services/stream/ContentBlock.kt
package com.jinloes.prpilot.services.stream

import kotlinx.serialization.Serializable

@Serializable
data class ContentBlock(
    val type: String? = null,
    val name: String? = null,
    val input: Map<String, @Serializable(with = AnySerializer::class) Any>? = null,
    val text: String? = null,
    val thinking: String? = null,
)
